src/git/from_vcs.py

```diff
@@ -111,18 +111,16 @@ def git_pieces_from_vcs(tag_prefix, root, verbose, runner=run_command):
         mo = re.search(r'^(.+)-(\d+)-g([0-9a-f]+)$', git_describe)
         if not mo:
             # unparsable. Maybe git-describe is misbehaving?
-            pieces["error"] = ("unable to parse git-describe output: '%s'"
-                               % describe_out)
+            pieces["error"] = f"unable to parse git-describe output: '{describe_out}'"
             return pieces
 
         # tag
         full_tag = mo.group(1)
         if not full_tag.startswith(tag_prefix):
+            error = f"tag '{full_tag}' doesn't start with prefix '{tag_prefix}'"
```

Comment:
No clear need for a temporary variable just to print and then assign...

```suggestion
            pieces["error"] = f"tag '{full_tag}' doesn't start with prefix '{tag_prefix}'"
```

Reply:
Indeed, this was for consistency with later fix.